### PR TITLE
Experience/9400/stlt dashboard api integration

### LINKED ISSUE #9400: STLT Data Dashboard - Hook up API to Data Dashboard UI
## Problem statement

As part of the STLT Dashboard, we need to implement the data of the main index page. 

## What you need to know

- Latest [Figma File](https://www.figma.com/proto/6mwI5ac6rprACKDzDo4Ady/ReportStream-Workspace-%7C-2023?type=design&node-id=6072-40941&t=q6sE1fXW050Nlyo7-1&scaling=min-zoom&page-id=2369%3A26962&starting-point-node-id=6072%3A40941&mode=design)
- Hook up to the API (PR #9720)

## Acceptance criteria

- [x] Data Dashboard should use updated API ([PR](https://github.com/CDCgov/prime-reportstream/pull/9720))

## To do

- [x] Update interface
- [x] Update hook
- [x] Update endpoint call in dataDashboard.ts
- [x] Hook up filter
- [x] Hook up table pagination
- [x] Unit tests

### PR DESCRIPTION
**Pull request by penny-lischer**
This PR ...

Hooks up the API to Data Dashboard UI

Test Steps:
1. Log into RS
2. Select an organization at the State level
3. Change URL to <server>/data-dashboard
4. View changes

## Changes
- Added API hook
- Added table pagination and sorting
- Hooked up the filter
- Added unit tests for DataDashboard and ReportDetails

Screenshot 2023-07-19 at 12 03 01 PM

### Testing
- [x] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [x] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

## Linked Issues
- Fixes #9400

### CODE REVIEW

File: frontend-react/cypress.config.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+"use strict";
```
Comment (KeithNava) [inline frontend-react/cypress.config.js:1]: [non-blocking] Do we need this as a `.js` file or is it something that comes dynamically in the build from the origin `.ts` file? or maybe we can turn the original file into a normal `.js` file? 🤔 ...just thinking cuz I've been looking at the cypress stuff for a bit

File: frontend-react/src/components/DataDashboard/DataDashboardTable/DataDashboardTable.tsx [hunk 2]
```diff
@@ -1,103 +1,113 @@
...
-}) => {
-    if (isLoading) return <Spinner />;
+    const {
+        data: results,
+        filterManager,
+        isLoading,
+    } = useReceiverDeliveries(activeService.name);
 
-    const featureEvent = `${FeatureName.DAILY_DATA} | ${EventName.TABLE_FILTER}`;
+    if (isLoading || !results) return <Spinner />;
```
Comment (KeithNava) [inline frontend-react/src/components/DataDashboard/DataDashboardTable/DataDashboardTable.tsx:44]: do we want to return a Spinner or a error message for no results? Maybe an opportunity to log?

Comment (penny-lischer) [reply frontend-react/src/components/DataDashboard/DataDashboardTable/DataDashboardTable.tsx:44]: This spinner will only appear when fetching the table data. If there are no results then the table component displays a message.

File: frontend-react/cypress.config.js (hunk 1 above)
Comment (penny-lischer) [reply frontend-react/cypress.config.js:1]: This file was added by mistake. Will remove

**Review by KeithNava — APPROVED** [review]

**Pull request merged**